packages/protocol/contracts/common/Accounts.sol

```diff
@@ -0,0 +1,483 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/utils/ReentrancyGuard.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+import "./interfaces/IAccounts.sol";
+
+import "../common/Initializable.sol";
+import "../common/Signatures.sol";
+import "../common/UsingRegistry.sol";
+
+contract Accounts is IAccounts, ReentrancyGuard, Initializable, UsingRegistry {
+
+  using SafeMath for uint256;
+
+  struct Signers {
+    // The address that is authorized to sign a vote on behalf of the account.
+    // The account can vote as well, whether or not an vote signing key has been specified.
+    address voting;
+    // The address that is authorized to sign consensus messages on behalf of the account.
+    // The account can manage the validator, whether or not an validation signing key has been
+    // specified. However if an validation signing key has been specified, only that key may
+    // actually participate in consensus.
+    address validating;
+
+    // The address of the key with which this account wants to sign attestations on the Attestations
+    // contract
+    address attesting;
+  }
+
+  struct Account {
+    bool exists;
+    // Each account may authorize signing keys to use for voting, valdiating or attestation.
+    // These keys may not be keys of other accounts, and may not be authorized by any other
+    // account for any purpose.
+    Signers signers;
+
+    // The address at which the account expects to receive transfers
+    address walletAddress;
+
+    string name;
+
+    // The ECDSA public key used to encrypt and decrypt data for this account
+    bytes dataEncryptionKey;
+
+    // The URL under which an account adds metadata and claims
```

Comment:
Comment should include an explanation or pointer to how this works

Reply:
A pointer to the docs will probably be ideal and currently planned for https://github.com/celo-org/celo-monorepo/issues/1077. I will update the reference here then